pkg/controlsvc/connect_test.go

```diff
@@ -0,0 +1,207 @@
+package controlsvc_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/ansible/receptor/pkg/controlsvc"
+	"github.com/ansible/receptor/pkg/controlsvc/mock_controlsvc"
+	"github.com/ansible/receptor/pkg/logger"
+	"github.com/golang/mock/gomock"
+)
+
+func CheckExpectedError(expectedError bool, errorMessage string, t *testing.T, err error) {
+	if expectedError {
+		if errorMessage != err.Error() {
+			t.Errorf("expected: %s , received: %s", errorMessage, err)
+		}
+	}
+
+	if !expectedError {
+		if err != nil {
+			t.Error(err)
+		}
+	}
+}
+
+func TestConnectInitFromString(t *testing.T) {
+	connectCommandType := controlsvc.ConnectCommandType{}
+
+	initFromStringTestCases := []struct {
+		name          string
+		expectedError bool
+		errorMessage  string
+		input         string
+	}{
+		{
+			name:          "no connect target",
+			expectedError: true,
+			errorMessage:  "no connect target",
+			input:         "",
+		},
+		{
+			name:          "too many parameters",
+			expectedError: true,
+			errorMessage:  "too many parameters",
+			input:         "one two three four",
+		},
+		{
+			name:          "three params - pass",
+			expectedError: false,
+			errorMessage:  "",
+			input:         "one two three",
+		},
+	}
+
+	for _, testCase := range initFromStringTestCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			_, err := connectCommandType.InitFromString(testCase.input)
+
+			CheckExpectedError(testCase.expectedError, testCase.errorMessage, t, err)
+		})
+	}
+}
+
+func TestConnectInitFromJSON(t *testing.T) {
+	connectCommandType := controlsvc.ConnectCommandType{}
+
+	initFromJSONTestCases := []struct {
```

Comment:
if you want, you can use the `InitFromJSONTestCase` struct that you defined in ping here. Im ok to leave it as is if you want